spec/index.bs

```diff
@@ -2517,6 +2561,112 @@ dictionary GPUImageBitmapCopyView {
+        bytesPerRowOfTexelBlocks * (copySize.height / blockHeight - 1) + (copySize.width / blockWidth * blockSize);
+    requiredBytesInCopy = bytesPerImage * (copySize.depth - 1) + bytesInLastSlice;
+    ```
+
+  The following validation rules apply:
+
+  For |encoder|:
+  - |encoder|.{{GPUCommandEncoder/copyBufferToTexture()}} must not be called when a {{GPURenderPassEncoder}}
+    is active on |encoder|.
+  - |encoder|.{{GPUCommandEncoder/copyBufferToTexture()}} must not be called when a {{GPUComputePassEncoder}}
+    is active on |encoder|.
+
+  For |source|:
+  - |source| must be [=valid=].
+  - |source|.{{GPUBufferCopyView/buffer}}.{{GPUBuffer/[[usage]]}} must contain {{GPUBufferUsage/COPY_SRC}}.
+
+  For |destination|:
+  - |destination| must be [=valid=].
+  - |destination|.{{GPUTextureCopyView/texture}}.{{GPUTexture/[[textureUsage]]}} must contain
+    {{GPUTextureUsage/COPY_DST}}.
+
+  For |copySize|:
+  - |copySize|.[=Extent3D/width=] cannot be 0.
+  - |copySize|.[=Extent3D/height=] cannot be 0.
+  - |copySize|.[=Extent3D/depth=] cannot be 0.
+  - If the {{GPUTexture/[[dimension]]}} of |destination|.{{GPUTextureCopyView/texture}} is
+    {{GPUTextureDimension/1d}}:
+    - Both |copySize|.[=Extent3D/height=] and [=Extent3D/depth=] must be 1.
+  - If the {{GPUTexture/[[dimension]]}} of |destination|.{{GPUTextureCopyView/texture}} is
+    {{GPUTextureDimension/2d}}:
+    - |copySize|.[=Extent3D/depth=] must be 1.
+
+  For the copy being in-bounds:
+  - If |source|.{{GPUBufferCopyView/rowsPerImage}} is not 0, it must be greater than or equal to
+    |copySize|.[=Extent3D/height=].
+  - (|source|.{{GPUBufferCopyView/offset}} &add; |required bytes in copy|) must not overflow a {{GPUSize64}}.
+  - (|source|.{{GPUBufferCopyView/offset}} &add; |required bytes in copy|) must be smaller than or equal to
+    |source|.{{GPUBufferCopyView/buffer}}.{{GPUBuffer/[[size]]}}.
+  - If the {{GPUTexture/[[dimension]]}} of |destination|.{{GPUTextureCopyView/texture}} is
+    {{GPUTextureDimension/2d}}:
+    - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/x}} &add; |copySize|.[=Extent3D/width=]) must be less than or equal to the [=Extent3D/width=] of the [=physical size=] of |destination|{{GPUTextureCopyView/texture}} [=subresource=] at [=mipmap level=] {{GPUTextureCopyView/mipLevel}}.
+    - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/y}} &add; |copySize|.[=Extent3D/height=]) must be less than or equal to the [=Extent3D/height=] of the [=physical size=] of |destination|.{{GPUTextureCopyView/texture}} [=subresource=] at [=mipmap level=] {{GPUTextureCopyView/mipLevel}}
+
+  For the texel block alignments:
+  - |source|.{{GPUBufferCopyView/rowsPerImage}} must be a multiple of |blockHeight|.
+  - |copySize|.[=Extent3D/width=] must be a multiple of |blockWidth|.
+  - |copySize|.[=Extent3D/height=] must be a multiple of |blockHeight|.
+  - |source|.{{GPUBufferCopyView/offset}} must be a multiple of |blockSize|.
```

Comment:
Hmm, I would prefer if we didn't have this. Does this restriction come from a backend? (D3D12?)

Reply:
Yes. It comes from [the "Valid Usage" of VkBufferImageCopy in Vulkan spec](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#copies-buffers-images):
"If the calling command’s VkImage parameter’s format is not a depth/stencil format or a multi-planar format, then bufferOffset must be a multiple of the format’s texel block size."